notification/source/cpp/src/KaaDemo.cpp

```diff
@@ -63,23 +63,72 @@ static std::string alertTypeToString(const kaa_notification::AlertType alertType
 
 class NotificationListener : public INotificationListener {
 public:
+    NotificationListener(std::shared_ptr<IKaaClient> kaaClient, std::atomic_bool &quitFlag):
+        kaaClient_(kaaClient), quitFlag_(quitFlag)
+    {}
+
     virtual void onNotification(const int64_t topicId, const KaaNotification &notification)
     {
-        std::cout << (boost::format("Notification for topic id '%1%' received") % topicId) << std::endl;
+        auto topics = kaaClient_->getTopics();
+        const auto &topic = std::find_if(topics.begin(), topics.end(),
+                [topicId](const Topic &t) { return t.id == topicId; });
+        std::cout << (boost::format("Notification for topic id '%1%' and name '%2%' received") % topicId % topic->name) << std::endl;
         std::cout << (boost::format("Alert type %1%, Alert message %2%") %
                 alertTypeToString(notification.alertType) %
                 (notification.alertMessage.empty() ? "Body is empty" : notification.alertMessage)) << std::endl;
+        if (topic->subscriptionType == SubscriptionType::OPTIONAL_SUBSCRIPTION) {
+            quitFlag_ = true;
+        }
     }
+
+private:
+    std::shared_ptr<IKaaClient> kaaClient_;
+    std::atomic_bool &quitFlag_;
 };
 
 class TopicListListener : public INotificationTopicListListener {
 public:
 
+    TopicListListener(std::shared_ptr<IKaaClient> kaaClient, std::atomic_bool &quitFlag):
+        kaaClient_(kaaClient), quitFlag_(quitFlag)
+    {}
+
     virtual void onListUpdated(const Topics& topics)
     {
-        std::cout << ("Topic list was updated") << std::endl;
+        std::cout << "Topic list was updated" << std::endl;
         showTopicList(topics);
         optionalTopicsList_ = extractOptionalTopicIds(topics);
+        for (;;) {
```

Comment:
It's just a nitpick, but `while (true)` it's a common way to express an infinite loop.